lib/ChainwebDb/Types/Event.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE ImpredicativeTypes #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module ChainwebDb.Types.Event where
+
+------------------------------------------------------------------------------
+import BasePrelude
+import Data.Aeson
+import Data.Text (Text)
+import Database.Beam
+import Database.Beam.Postgres (PgJSONB)
+------------------------------------------------------------------------------
+import ChainwebDb.Types.Transaction
+------------------------------------------------------------------------------
+
+
+------------------------------------------------------------------------------
+data EventT f = Event
+  { _ev_requestKey :: PrimaryKey TransactionT f
```

Comment:
@sirlensalot I think this has problems for events that aren't attached to a transaction.  (The main event in question here is coinbase events.)  It seems like we need to modify this event table schema to allow this.  Any thoughts about a good way to do this?  One idea that comes to mind is adding a `blockHash` field and making the `requestKey` field a `Maybe`.

Another factor along these lines to consider is a mechanism to inherently track the progress of event backfill.  The other PR introduces a `numEvents` field into the transactions table to facilitate this, but we also need to consider the case where a block has no transactions (and thus no `numEvents` field) but we still need to track progress backfilling the coinbase events.  Any thoughts on this?

Reply:
I don't think the request key should be a maybe type. Coinbase events still have request keys within them. Unless there is something that I'm missing and there are other types of events without request keys?